### ISSUE TITLE
PanicException when using `.to_parquet()` with `use_pyarrow=False`

### ISSUE THREAD

**Issue by DrMaphuse** [pola-rs/polars#2861]
#### What language are you using?

**Python**

#### What version of polars are you using?

0.13.5

#### What operating system are you using polars on?

Ubuntu 20.04

#### What language version are you using
Python 3.9.7

#### Describe your bug.

`df.to_parquet()` throws an Exception on larger dataframes with null values in `int` or `bool`-columns:

```python
PanicException: called `Result::unwrap()` on an `Err` value: InvalidArgumentError("Chunk require all its arrays to have an equal number of rows")
```

#### What are the steps to reproduce the behavior?

Impossible to reproduce (see below).

#### What is the actual behavior?

* `df.to_parquet()` produces a PanicException.
* Sample data does not produce the abovementioned PanicException
* Saving to csv, reading back to polars, and saving back to parquet works fine without producing the error
* `use_pyarrow=True` fixes the error, but takes significantly longer
* Changing compression does not seem to fix the error

#### What is the expected behavior?

No error.

**Comment by ritchie46** [comment]
> Impossible to reproduce (see below).

What do you mean?

**Comment by DrMaphuse** [comment]
Hi! I get this error simply when trying to write a dataframe to a parquet file using `to_parquet()`, but I cannot create a sample dataframe that produces this same error.

This code will produce a very similar dataframe to the one I have. The exception occurs after the string modification.
```
import random
import polars as pl
n = 10000000
l1 = pl.Series(random.choices(range(100000, 999999), k=int(n/10)) + [None]*int(n/10*9)).shuffle()
l2 = pl.Series(random.choices(['A_b', None], k=n))
df = pl.DataFrame([l1, l2])
df = df.with_column(pl.col('column_1').str.to_lowercase().str.replace("_", " "))
df.to_parquet("test.parquet")
```
But this sample does not not actually reproduce the error. It just serves to give an idea of the steps I take in order to produce the error on my end.

**Comment by DrMaphuse** [comment]
Full traceback might give additional pointers:

```python
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidArgumentError("Chunk require all its arrays to have an equal number of rows")', /github/home/.cargo/git/checkouts/arrow2-8a2ad61d97265680/22bf173/src/chunk.rs:19:31

---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
/tmp/ipykernel_138401/1692828431.py in <module>
----> 1 pv.to_parquet(out_folder / "test.parquet")

~/.local/lib/python3.9/site-packages/polars/internals/frame.py in to_parquet(self, file, compression, statistics, use_pyarrow, **kwargs)
   1155             )
   1156         else:
-> 1157             self._df.to_parquet(file, compression, statistics)
   1158 
   1159     def to_numpy(self) -> np.ndarray:

PanicException: called `Result::unwrap()` on an `Err` value: InvalidArgumentError("Chunk require all its arrays to have an equal number of rows")
```

**Comment by ghuls** [comment]
This should fix it:
```python
df.rechunk().to_parquet("test.parquet")
```

Your dataframe very likely has a diferent number of chunks for each column:

```pyhon
In [114]: df.n_chunks()
Out[114]: 1

In [115]: df.column_0.n_chunks()
/home/luna.kuleuven.be/u0079808/software/anaconda3/envs/polars/lib/python3.8/site-packages/polars/internals/frame.py:1254: UserWarning: accessing series as Attribute of a DataFrame is deprecated
  warnings.warn("accessing series as Attribute of a DataFrame is deprecated")
Out[115]: 1

In [116]: df.column_1.n_chunks()
/home/luna.kuleuven.be/u0079808/software/anaconda3/envs/polars/lib/python3.8/site-packages/polars/internals/frame.py:1254: UserWarning: accessing series as Attribute of a DataFrame is deprecated
  warnings.warn("accessing series as Attribute of a DataFrame is deprecated")
Out[116]: 2
```

@ritchie46 Seems like df.n_chunks does not check the number of chunks for each column.

**Comment by DrMaphuse** [comment]
> df.rechunk().to_parquet("test.parquet")

This worked, but it doubled the time to save, making it quite similar to saving with pyarrow. 

On another note, is there any way to know where/why the chunks became uneven?

**Comment by ghuls** [comment]
> On another note, is there any way to know where/why the chunks became uneven?
it depends on the operations that you did before. Could you paste the full ocde?

If you would to the rechunk on only the series itself (that has multiple chunks) before creating the dataframe from individual series, the rechunk will be much faster as it doesn' need to rechunk all other series which already have 1 chunk.

**Comment by ritchie46** [comment]
Fixed by: #2875
